tempColorTokenBuild.ts

```diff
@@ -0,0 +1,172 @@
+    ]
+  ]
+
+  const getStyleDictionaryConfig: StyleDictionaryConfigGenerator = (
+    outputName,
+    source,
+    include,
+    options
+  ): StyleDictionary.Config => ({
+    source, // build the special formats
+    include,
+    parsers: [w3cJsonParser],
+    // formats need to be defined here but are used in platforms
+    format: {
+      'scss/mixin-css-variables': scssMixinCssVariables,
+      'javascript/esm': javascriptEsm,
+      'javascript/commonJs': javascriptCommonJs,
+      'typescript/export-definition': typescriptExportDefinition
+    },
+    // transforms need to be defined here but are used in platforms
+    transform: {
+      'color/rgbAlpha': colorToRgbAlpha,
+      'color/hexAlpha': colorToHexAlpha,
+      'color/hex6': colorToHex6
+    },
+    platforms: {
+      css: platformCss(`${outputName}.css`, options.prefix, options.buildPath),
+      scss: platformScss(`${outputName}.scss`, options.prefix, options.buildPath),
+      js: platformJs(`${outputName}.js`, options.prefix, options.buildPath),
+      ts: platformTs(`${outputName}.ts`, options.prefix, options.buildPath),
+      json: platformJson(`${outputName}.json`, options.prefix, options.buildPath),
+      docJson: platformDocJson(`${outputName}.json`, options.prefix, options.buildPath)
+    }
+  })
+
+  /**
+   * Copy `removed` files
+   */
+  copyFilesAndFolders([`removed`], `tokens`, buildOptions.buildPath)
+
+  /**
+   * convert colors
+   */
+  for (const [theme, source, include] of themes) {
+    StyleDictionary.extend(
+      getStyleDictionaryConfig(`color/${theme}`, source, include, buildOptions)
+    ).buildAllPlatforms()
+  }
+  // TODO: Remove once shadows that used to be in colors are implemented
+  // eslint-disable-next-line no-console
+  console.log('⚠️ Shadows are not implemented in tokens correctly')
+  /**
+   * build deprecated.json
+   */
+  const deprecatedBuilds: [string, string[], string[]][] = [
+    ['color', themes[0][1], themes[0][2]] // light mode
+  ]
+  // get config
+  const deprecatedConfig: StyleDictionaryConfigGenerator = (outputName, source, include, options) => ({
+    source,
+    include,
+    parsers: [w3cJsonParser],
+    format: {
+      'typescript/export-definition': typescriptExportDefinition
+    },
+    transform: {
+      'color/hexAlpha': colorToHexAlpha,
+      'color/hex6': colorToHex6,
+      'json/deprecated': jsonDeprecated,
+      'name/pathToDotNotation': namePathToDotNotation
+    },
+    platforms: {
+      deprecated: platformDeprecatedJson(`${outputName}.json`, options.prefix, options.buildPath),
+      types: platformTypeDefinitions(`${outputName}`, options.prefix, options.buildPath)
+    }
+  })
+  //
+  for (const [name, source, include] of deprecatedBuilds) {
+    //
+    StyleDictionary.extend(deprecatedConfig(name, source, include, buildOptions)).buildAllPlatforms()
+  }
+}
+
+buildDesignTokens({
+  buildPath: 'tokens-v2-private',
+  prefix: undefined
```

Comment:
Should this be 'primer'?

Reply:
Also, on the build path, I'm guessing it's intentional but the colors output in a different folder to the other existing v2 tokens. 
<img width="587" alt="Screenshot 2022-10-12 at 15 36 58" src="https://user-images.githubusercontent.com/13340707/195374623-a1705afa-8ca2-45a2-9f36-317f8457b2ca.png">

Reply:
Yes, this is intentional. I see no reason why we should have a another level of nesting and @langermank didn't reply / disagree when I mentioned to remove this somewhere a few weeks ago. Ultimately we should move all tokens out of the tokens subfolder.

> Should this be 'primer'?
not for not, to keep parity, we have no prefix anywhere.

Reply:
> Not for now, to keep parity, we have no prefix anywhere.

Oh i see. I thought this was one of the main distinctions with v2 tokens, that we always namespace them. It was also to help avoid confusion when brand tokens are available alongside the the product tokens.

Reply:
I am not sure, I am all in for doing this. If we want to, we can just add it here. 

I am fine either way, it will just mean **all tokens** don't match and **all** will have to be replaced. If this is a definite, adding it in is literally changing this one line of code.

I think this should not be part of this PR as it is a longer discussion including if `ts` and `js` also have this because you import from something and name it again.

I believe for this PR (as it does not produce any output now) it is enough that the possibility is available to make it work either way.

Do you agree?

Reply:
> tokens sub folder

I'm fine with removing this `tokens` sub folder, but I think base scales should be inside `/base` and functional colors should be inside `/functional`. I think you're building all themes (base scale and functional) into a single file, which I'm not sure about.. it feels like they should be separated the same way the base/functional sizing tokens are.

> primer prefix

I'm okay with leaving it off for now since we're not shipping these tokens yet, but it will be required for all the renames.

Reply:
> I don't consider this merge blocking so long as we can keep iterating from here 👍

Yes, we can iterate on this until it is perfect.

> I'm okay with leaving it off for now since we're not shipping these tokens yet, but it will be required for all the renames.

I understood this a lot better now in our zoom and will work on getting the naming right in a new PR were we can only focus on `folder structure` and `naming/prefix`.